File: bank-service/src/main/java/rs/edu/raf/service/impl/CreditScheduler.java
package rs.edu.raf.service.impl;

import lombok.AllArgsConstructor;
import org.springframework.scheduling.annotation.Scheduled;
import org.springframework.stereotype.Component;
import rs.edu.raf.service.CreditService;

@Component
@AllArgsConstructor
public class CreditScheduler {
    private CreditService creditService;

    @Scheduled(cron = "0 0 0 * * ?")
    public void processCreditInstallments() {
        creditService.processCreditInstallment();
    }
}

File: bank-service-kotlin/src/main/kotlin/org/example/bankservicekotlin/service/impl/CreditScheduler.kt
package org.example.bankservicekotlin.service.impl

import org.example.bankservicekotlin.service.CreditService
import org.springframework.scheduling.annotation.Scheduled
import org.springframework.stereotype.Component

@Component
class CreditScheduler(private val creditService: CreditService) {
    @Scheduled(cron = "0 0 0 * * ?")
    fun processCreditInstallments() {
        creditService.processCreditInstallment()
    }

}